src/ert/storage/everest_ensemble.py

```diff
@@ -0,0 +1,37 @@
+from pydantic import BaseModel
+from typing_extensions import TypedDict
+
+from .local_ensemble import LocalEnsemble
+
+
+class EverestRealizationInfo(TypedDict):
+    geo_realization: int
+    perturbation: int | None  # None means it stems from unperturbed controls
+    # Q: Maybe we also need result ID, or no? Ref if we have multiple evaluations
+    # for unperturbed values, though the ERT real id will also differentiate them
+
+
+class _Index(BaseModel):
+    ert2ev_realization_mapping: dict[int, EverestRealizationInfo] | None = None
+
+
+class EverestEnsemble:
+    def __init__(self, ert_ensemble: LocalEnsemble):
+        self._ert_ensemble = ert_ensemble
+        self._index = _Index()
+
+    @property
+    def ert_ensemble(self) -> LocalEnsemble:
+        return self._ert_ensemble
+
+    def save_realization_mapping(
+        self, realization_mapping: dict[int, EverestRealizationInfo]
+    ) -> None:
+        self._index.ert2ev_realization_mapping = realization_mapping
+        self._ert_ensemble._storage._write_transaction(
+            self.ert_ensemble._path / "everest_index.json",
```

Comment:
Is this needed? Could we integrate this more closely into the regular `Ensemble`? No reason that could not have metadata